cost/aws/savings_plan/expiration/aws_savings_plan_expiration.pt

```diff
@@ -0,0 +1,175 @@
+name "AWS Expiring Savings Plans"
+rs_pt_ver 20180301
+type "policy"
+short_description "A policy that sends email notifications before AWS Savings Plan expire. See the [README](https://github.com/rightscale/policy_templates/tree/master/cost/aws/reserved_instances/expiration) and [docs.flexera.com/flexera/EN/Automation](https://docs.flexera.com/flexera/EN/Automation/AutomationGS.htm) to learn more."
+long_description ""
+severity "medium"
+category "Cost"
+tenancy "single"
+default_frequency "daily"
+info(
+  version: "2.0",
+  provider: "Flexera Optima",
+  service: "",
+  policy_set: ""
+)
+
+###############################################################################
+# User inputs
+###############################################################################
+
+parameter "param_heads_up_days" do
+  type "number"
+  label "Number of days to prior to expiration date to trigger incident"
+  default 7
+end
+
+parameter "param_email" do
+  type "list"
+  label "Email addresses of the recipients you wish to notify"
+end
+
+###############################################################################
+# Authentication
+###############################################################################
+
+#AUTHENTICATE WITH AWS
+credentials "auth_aws" do
+  schemes "aws","aws_sts"
+  label "AWS"
+  description "Select the AWS Credential from the list"
+  tags "provider=aws"
+end
+
+###############################################################################
+# Datasources
+###############################################################################
+
+datasource "ds_savings_plans" do
+  request do
+    auth $auth_aws
+    host "savingsplans.us-east-1.amazonaws.com"
+    verb "POST"
+    path "/DescribeSavingsPlans"
+    header "User-Agent", "RS Policies"
+    header "Content-Type", "application/json"
+  end
+  result do
+    encoding "json"
+    collect jmes_path(response, "savingsPlans[*]") do
+      field "savingsPlanID", jmes_path(col_item, "savingsPlanId")
+      field "savingsPlanArn", jmes_path(col_item, "savingsPlanArn")
+      field "state", jmes_path(col_item, "state")
+      field "tags", jmes_path(col_item, "tags")
+      field "termDurationInSeconds", jmes_path(col_item, "termDurationInSeconds")
+      field "end", jmes_path(col_item, "end")
+      field "start", jmes_path(col_item, "start")
+      field "commitment", jmes_path(col_item, "commitment")
+      field "offeringID", jmes_path(col_item, "offeringId")
+    end
+  end
+end
+
+datasource "ds_filter_savings_plans" do
+  run_script $js_filter_savings_plans, $ds_savings_plans, $param_heads_up_days
+end
+###############################################################################
+# Scripts
+###############################################################################
+
+script "js_describe_savings_plans", type: "javascript" do
```

Comment:
Cannot see where this script is referenced by a datasource. Doesn't break the template but just thought it was worth mentioning.